### ISSUE TITLE
SelectOneListbox: NVDA screenreader issue with Firefox

### ISSUE THREAD

**Issue by f1nnk** [primefaces/primefaces#11623]
### Describe the bug

Similar to [this](https://github.com/primefaces/primefaces/issues/11577) issue i found, the NVDA screenreader doesn't work correctly for the SelectOneListbox in Firefox when navigating with the keyboard. The screenreader only reads out "selected" or "not selected" instead of the elements. This issue doesn't exist in other browsers like Chrome.
It does work with another similar listbox I've tried though: https://primeng.org/listbox
But I'm not sure if its even a PrimeFaces issue of if the cause lies elsewhere. As you mentioned in the linked issue, the problem doesn't seem to be that troublesome but I just wanted to let you know it exists.

### Reproducer

Navigate through the SelectOneListbox with the keyboard (arrow keys) in the [Showcase](http://www.primefaces.org/showcase/ui/input/listbox.xhtml?jfwid=4576d) with NVDA and Firefox

### Expected behavior

NVDA should read out the selected elements

### PrimeFaces edition

None

### PrimeFaces version

Showcase version

### JSF implementation

All

### JSF version

/

### Java version

/

### Browser(s)

Firefox 123.0.1 (64-Bit)

**Comment by melloware** [comment]
Is there any chance you can open an NVDA ticket and link this ticket to it?  I have a feeling its a bug in NVDA if it works in Chrome and with other ScreenReaders?

**Comment by melloware** [comment]
Reported to NVDA: https://github.com/nvaccess/nvda/issues/16300

**Comment by melloware** [comment]
Tracking in NVDA ticket.

**Comment by f1nnk** [comment]
@melloware Yes for whatever reason you can tab into the listbox twice in Firefox with Advanced mode. Tabbing into it once (disregarding the filter) should be the only way to navigate the listbox. NVDA reads "selected" when selecting the selected element and "not selected" when moving to the element above or below it. For the rest of the elements it doesn't read out anything. With left and right arrow keys it works though (but of course the listbox and the selected element don't change).

When tabbing into it twice, NVDA works with the up and down arrow keys and you can select elements, but you can navigate further into the DOM past the listbox and it doesn't change visually. That only worked in my case though and not in the [Showcase](http://www.primefaces.org/showcase/ui/input/listbox.xhtml?jfwid=2f118)

**Comment by melloware** [comment]
@f1nnk i submitted a PR that improves the reader for "Item 4 of 10" etc.   I think what is different is PF uses a `<table>` to represent the Advanced mode and PrimeNG does not.  But I still think the real issue is NVDA support in Firefox.  The ARIA tags we are using are all the correct ones and as you mentioned it reads perfectly fine in Chrome.

**Comment by f1nnk** [comment]
@melloware yea i agree. What does your change do exactly? I guess you can close this issue again then 😄

**Comment by melloware** [comment]
OK so what I added was:

1. Handle ARROW_RIGHT and ARROW_LEFT so the behavior is not different.
2. Added `aria-activedescendant` which is what the PrimeNG showcase does on the listbox to let it know what option is highlighted.
3. Added 'aria-setSize` and `aria-posinset` so it reads "4 of 9" etc.
